### ISSUE TITLE
[store/meta]: add disk backed raft-state impl

### ISSUE THREAD

**Issue by drmingdrmer** [datafuselabs/datafuse#1051]
**Summary**

Persist raft state, which includes id, `current_term` and `voted_for` on disk.

This is the first step to impl the disk backed raft meta.
The next step is to impl disk-backed log and state machine.

Part of #271

**Comment by jovany-wang** [comment]
Sorry for that I didn't understand the purpose to persist the metas.  
In general cases, it's totally enough to persist log entries and user-defined state machines. Is there any context I have missed?

**Comment by drmingdrmer** [comment]
There are some other information must be persisted, according to the raft paper, including the current term, and the leader a node has voted for.

E.g. if the `voted_for` get lost when restarting a raft node, there will be two nodes believes they are the legal leader.
Then the log get messed up:

```
R1 elect       become_leader                                  write_log1
   |                                                          |
   v                                                          v
R2 vote_for_R1               crash vote_for_R3                log1?log2?
                                   ^                          ^
                                   |                          |
R3                                 elect        become_leader write_log2
-------------------------------------------------------> time
```

**Comment by jovany-wang** [comment]
> There are some other information must be persisted, according to the raft paper, including the current term, and the leader a node has voted for.
> 
> E.g. if the `voted_for` get lost when restarting a raft node, there will be two nodes believes they are the legal leader.
> Then the log get messed up:
> 
> ```
> R1 elect       become_leader                                  write_log1
>    |                                                          |
>    v                                                          v
> R2 vote_for_R1               crash vote_for_R3                log1?log2?
>                                    ^                          ^
>                                    |                          |
> R3                                 elect        become_leader write_log2
> -------------------------------------------------------> time
> ```

OK, I see.

**Comment by drmingdrmer** [comment]
closed by #1056
